core/sr-arithmetic/fuzzer/src/rational128.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Running
+//! Running this fuzzer can be done with `cargo hfuzz run rational128`. `honggfuzz` CLI options can
+//! be used by setting `HFUZZ_RUN_ARGS`, such as `-n 4` to use 4 threads.
+//!
+//! # Debugging a panic
+//! Once a panic is found, it can be debugged with
+//! `cargo hfuzz run-debug rational128 hfuzz_workspace/rational128/*.fuzz`.
+//!
+//! # More infomation
+//! More information about `honggfuzz` can be found
+//! [here](https://docs.rs/honggfuzz/0.5.45/honggfuzz/).
+
+use honggfuzz::fuzz;
+use sr_arithmetic::{helpers_128bit::multiply_by_rational, traits::Zero};
+
+fn main() {
+	loop {
+		fuzz!(|data: ([u8; 16], [u8; 16], [u8; 16])| {
+			let (a_bytes, b_bytes, c_bytes) = data;
+			let (a, b, c) = (
+				u128::from_be_bytes(a_bytes),
+				u128::from_be_bytes(b_bytes),
+				u128::from_be_bytes(c_bytes),
+			);
+
+			println!("++ Equation: {} * {} / {}", a, b, c);
+
+			let truth = mul_div(a, b, c);
+			// if Err just use the truth value. We don't care about such cases. The point of this
+			// fuzzing is to make sure: if `multiply_by_rational` returns `Ok`, it must be 100% accurate
+			// returning `Err` is fine.
+			let result = multiply_by_rational(a, b, c).unwrap_or(truth);
```

Comment:
Oh, my bad. I think I should have made the doc more accurate. The new implementation of `multiply_by_rational ` should _always return a 100& accurate value as long as it fits in u128_. So this must change a bit and unwrap_or() is not correct

Reply:
And I suppose about the +- 1problem you can ignore it 🤔

Reply:
?